### ISSUE TITLE
00_decode_fix repeating group assertions failing

### ISSUE THREAD

**Issue by surrsurus** [ferrumfix/ferrumfix#12]
**Describe the bug**
Hi, currently looking into FIX serializers/deserializers in rust and was messing around with the examples. Noticed that the assertions on the repeating groups all fail. The FIX header parsing seems to work fine, and as soon as it starts trying to look at

```rust
  for entry in md_entries.entries() {
        assert_eq!(entry.fv(fix42::CURRENCY), Ok(b"EUR"));
    }
```

Every assertion including the above start turning up with `Err(None)`. Running this on rust stable.

**To Reproduce**
1. Cd into 00_decode_fix
2. `cargo run`

**Expected behavior**
The assertions pass

**Screenshots**
Logs:

```
    Finished dev [unoptimized + debuginfo] target(s) in 0.19s
     Running `/Users/gray/bin/wolf/ferrumfix/target/debug/example_decode_fix`
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `Err(None)`,
 right: `Ok([69, 85, 82])`', examples/00_decode_fix/src/main.rs:41:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace```

**Comment by surrsurus** [comment]
Ah I also forgot you will also need the Cargo.toml fix which is:
```rust
[dependencies]
fefix = { path = "../../crates/fefix", features = ["fix42"] }
```
Basically how you have it in the other examples.

**Comment by Osse** [comment]
I stumbled across this as well. There seems to be a bug in the repo group handling. If you iterate over the entries() of a group instance then it iterates over the whole message.

**Comment by Osse** [comment]
It seems to be a problem with the FIX 4.2 dictionary. By changing the replacing it with `Dictionary::fix44()` the asserts pass. However some later ones still fail, and the `len()` of each entry is 24 (same as the whole message) instead of 8. Not sure if that's intended or not.

**Comment by neysofu** [comment]
This quite a subtle bug and, as far as I can tell, only happens in <= FIX 4.2. FIX 4.3 introduces the `NumInGroup` FIX datatype, which is how FerrumFIX recognizes groups. As a result, repeating groups are not detected at all for all messages with <= FIX 4.2.

As a temporary workaround, I'm adding some field name parsing logic: all fields that start with `No` or end with `Len` are automatically flagged as group tags. This is far from an ideal solution, but it fixes the example and - I suspect - succeeds in handling a huge number of groups for old FIX versions, so I'm going with that for now. I'll have to think about a better long-term solution.

Thanks for the bug report!
